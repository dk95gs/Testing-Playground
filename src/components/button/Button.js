import React from 'react';
import { render } from '@testing-library/react';

const Button = (props) =>{
    const styles = {
        width:"20rem",
        height:"4rem",
        fontSize:"2rem",
        backgroundColor:"blue",
        opacity: ".7",
        borderRadius:"25px",
        outline:"none",
        boxShadow:"none"
    }
    console.log(props.DT);
    return(
        <div>
            <button data-testid="button" onClick={()=>{props.setDT(props.DT * 2)}} style={styles}>{props.name}</button>
        </div>
    )
}

export default Button;