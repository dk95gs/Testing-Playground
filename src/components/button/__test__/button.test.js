import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import {render, cleanup} from '@testing-library/react';
import reactTestRender from 'react-test-renderer';

import "@testing-library/jest-dom/extend-expect";


afterEach(cleanup);

it("renders without crashing", () =>{
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
})

it("renders button correctly", () =>{
    const {getByTestId} = render(<Button name="Hello"/>);
    expect(getByTestId('button')).toHaveTextContent("Hello");
})


it("renders button correctly", () =>{
    const {getByTestId} = render(<Button name="Good Bye"/>);
    expect(getByTestId('button')).toHaveTextContent("Good Bye");
})

it("matches snapshop", () =>{
    const tree = reactTestRender.create(<Button name="Hello"/>).toJSON();
    expect(tree).toMatchSnapshot();
})
