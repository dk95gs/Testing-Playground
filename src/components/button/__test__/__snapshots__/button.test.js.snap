// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<div>
  <button
    data-testid="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "borderRadius": "25px",
        "boxShadow": "none",
        "fontSize": "2rem",
        "height": "4rem",
        "opacity": ".7",
        "outline": "none",
        "width": "20rem",
      }
    }
  >
    Hello
  </button>
</div>
`;
